trieste/models/model_interfaces.py

```diff
@@ -504,29 +504,47 @@ class SparseVariational(GPflowPredictor, TrainableProbabilisticModel):
     A :class:`TrainableProbabilisticModel` wrapper for a GPflow :class:`~gpflow.models.SVGP`.
     """
 
-    def __init__(self, model: SVGP, data: Dataset, optimizer: Optimizer | None = None):
+    def __init__(
+        self,
+        model: SVGP,
+        optimizer: Optimizer | None = None
+    ):
         """
         :param model: The underlying GPflow sparse variational model.
-        :param data: The initial training data.
         :param optimizer: The optimizer with which to train the model. Defaults to
-            :class:`~trieste.models.optimizer.Optimizer` with :class:`~gpflow.optimizers.Scipy`.
+            :class:`~trieste.models.optimizer.Optimizer` with :class:`~tf.optimizers.Adam`.
+        :param max_num_inducing_points: The maximum number of inducing points allowed.
         """
+
+        if optimizer is None:
+            optimizer = Optimizer(tf.optimizers.Adam())
+
         super().__init__(optimizer)
         self._model = model
-        self._data = data
 
     def __repr__(self) -> str:
         """"""
-        return f"SparseVariational({self._model!r}, {self._data!r}, {self.optimizer!r})"
+        return f"SparseVariational({self._model!r}, {self.optimizer!r})"
 
     @property
     def model(self) -> SVGP:
         return self._model
 
     def update(self, dataset: Dataset) -> None:
-        _assert_data_is_compatible(dataset, self._data)
+        # Hard-code asserts from _assert_data_is_compatible because model doesn't store dataset
```

Comment:
this stuff looks like it might be useful to write a test to catch the issue that happened with the previous implementation